examples/grafana-agent-auto-instrumentation/java/README.md

```diff
@@ -0,0 +1,46 @@
+# Grafana Agent Java profiling via auto-instrumentation example
+
+This repository provides a practical demonstration of leveraging the Grafana Agent for continuous Java application profiling using Pyroscope in a dockerized environment. It illustrates a seamless approach to profiling Java processes, aiding in performance optimization.
+
+## Overview
+
+Using this example, you'll learn how to use Grafana Agent for continuous Java process profiling using Pyroscope with the async-profiler. This example provides an insightful view into application performance.
+
+### Profiling methodologies
+
+Java applications can be profiled via Pyroscope using three methodologies. The example emphasizes the first method, showcasing the attachment of the Pyroscope profiler as a `javaagent`. 
+
+Refer to the [official documentation](https://grafana.com/docs/pyroscope/latest/configure-client/grafana-agent/java/) for an in-depth understanding and additional configuration options for Java profiling with the Grafana Agent.
+
+The three methodologies are as follows:
+
+1. **Java Agent** (this example): Attach the Pyroscope profiler as a `javaagent` to a running process, initializing the profiler at application startup without source code modifications or extra dependencies. Manage configuration using Java parameters or environment variables.
+
+2. **OTel Integration** ([docs](https://grafana.com/docs/pyroscope/next/configure-client/trace-span-profiles/java-span-profiles/)): Utilize the OTel integration as an extension to `otel-java-instrumentation`, in conjunction with `opentelemetry-javaagent` to link java profiles with trace spans. This method is suitable for applications already using OpenTelemetry.
+
+3. **Direct Dependency** ([docs](https://grafana.com/docs/pyroscope/next/configure-client/language-sdks/java/)): Integrate Pyroscope directly within your application's code to allow dynamic profiler management.
```

Comment:
This example doesn't show the first method (or the other 2). In this approach the grafana agent attaches to the running Java processes and auto-instruments them (starts profiling) with async-profiler. Perhaps remove this section altogether and add a note in the next section about how this works. 

To clarify, the first method mentioned above requires changing the command to start the java application, an example of this is https://github.com/grafana/pyroscope/blob/94c1b653676fae43da8bddec2ed866a1ce19a3ae/examples/language-sdk-instrumentation/java/rideshare/Dockerfile#L40-L42